src/components/Modals/ModalPost/ModalPost.tsx

```diff
@@ -1,72 +1,149 @@
+import { useAppDispatch, useAppSelector } from '../../../redux/hooks';
+import { getUser, getModal, getPosts } from '../../../redux/selectors';
 import { Modal } from '../../Modal/Modal';
-import { ModalPostProps } from '../../../types/post';
+import { newPostData, updatedPostData } from '../../../types/post';
 import Author from '../../Author/Author';
 import Likes from '../../Likes/Likes';
-import { addPost, updatePost } from '../../../redux/postSlice';
-
-// TODO: correct date format in post__date
-
-function ModalPost({
-  id,
-  text,
-  photoUrl,
-  authors,
-  likes,
-  createdAt,
-  onClose,
-}: ModalPostProps) {
+import { addPost, editPost, removePost } from '../../../redux/postSlice';
+import { formatTime } from '../../../utils/helpers';
+import {
+  createPost,
+  deletePosts,
+  getUserInfo,
+  updatePost,
+} from '../../../utils/api';
+import { setUser } from '../../../redux/userSlice';
+import { useImageUrl } from '../../../hooks/useImageUrl';
+import imgPlaceholder from '../../../assets/post-placeholder.jpg';
+
+type ModalPostProps = {
+  postId: string;
+  onClose: () => void;
+  newPost?: {
+    text: string;
+    photoUrl: string;
+    authors: {
+      host: {
+        _id: string;
+        name: string;
+        avatar: string;
+      };
+      resident: {
+        _id: string;
+        name: string;
+        avatar: string;
+        species: string;
+      };
+    };
+    createdAt: string;
+    likes: [];
+  };
+};
+
+function ModalPost({ postId, onClose, newPost }: ModalPostProps) {
   const navigate = useNavigate();
-  const dispatch = useDispatch();
-  const currentUser = useSelector(getUser);
-  const modalIsActive = useSelector(getModal);
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(getUser);
+  const posts = useAppSelector(getPosts);
+  const modal = useAppSelector(getModal);
+  const [postText, setPostText] = useState('');
   const [postModalMode, setPostModalMode] = useState(
-    modalIsActive === 'add-post-next' ? 'edit' : 'view'
+    modal.modalIsActive === 'add-post-next' ? 'edit' : 'view'
   );
-  const [postText, setPostText] = useState(text);
+
+  const post = newPost || posts.find((post) => post._id === postId);
+
+  useEffect(() => {
+    if (post) {
+      setPostText(post.text);
+    }
+  }, [post]);
+
+  if (!post) {
+    return <p>Post not found</p>;
+  }
+
+  const { photoUrl, authors, likes, createdAt } = post;
+
+  const postingTime = formatTime(new Date(createdAt));
+  const isToday =
+    createdAt.slice(0, 10) === new Date().toISOString().slice(0, 10);
 
   const handleEditClick = () => {
-    console.log('edit');
     setPostModalMode('edit');
   };
 
   const handleTextChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
     setPostText(e.target.value);
   };
 
+  const handleDeleteClick = () => {
+    deletePosts(localStorage.jwt, postId).then(() => {
+      dispatch(removePost({ _id: postId }));
+      onClose();
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**

Reply:
Sorry, I didn't notice all the cases. I'll double-check again